pymatgen/io/vasp/inputs.py

```diff
@@ -2377,24 +2387,24 @@ def _gen_potcar_summary_stats(
     """
     func_dir_exist: dict[str, str] = {}
     vasp_psp_dir = vasp_psp_dir or SETTINGS.get("PMG_VASP_PSP_DIR")
-    for func in PotcarSingle.functional_dir:
-        cpsp_dir = f"{vasp_psp_dir}/{PotcarSingle.functional_dir[func]}"
+    for func, func_dir in PotcarSingle.functional_dir.items():
+        cpsp_dir = f"{vasp_psp_dir}/{func_dir}"
         if os.path.isdir(cpsp_dir):
-            func_dir_exist[func] = PotcarSingle.functional_dir[func]
+            func_dir_exist[func] = func_dir
         else:
-            warnings.warn(f"missing {PotcarSingle.functional_dir[func]} POTCAR directory")
+            warnings.warn(f"missing {func_dir} POTCAR directory")
```

Comment:
> @DanielYang59, see my comment above, these tests pass outside this PR. They're failing because of changes made by this PR that are easy to fix - opened a PR on your fork to fix it.

@esoteric-ephemera Thanks a lot for pointing out! I didn't even realize its my changes breaking the unit tests. (Didn't see the comments for some reason). But I fixed it here (removed previous re-assignment of `casp_dir` to `func_dir`).

@janosh Should be ready to merge. Thanks a lot.